### DISCUSSION TITLE
Which files can be safely removed from the x64 portable version if not using the shell extension?

### DISCUSSION THREAD

**Discussion by Gitoffthelawn** [WinMerge/winmerge#1091] [Q&A] [answered]
There's a long "why" to this, but I won't bore you with the lengthy details.

Which files can be safely removed from the x64 portable version of WinMerge if not using the shell extension (without causing any internal or external errors or performance issues)?

- [ ] Register.bat
- [ ] RegisterPerUser.bat
- [ ] RegisterWinMergeContextMenuPackage.bat
- [ ] ShellExtensionU.dll
- [ ] ShellExtensionX64.dll
- [ ] UnRegister.bat
- [ ] UnRegisterPerUser.bat
- [ ] UnregisterWinMergeContextMenuPackage.bat
- [ ] vcomp140.dll
- [ ] WinMerge32BitPluginProxy.exe
- [ ] WinMergeContextMenu.dll
- [ ] WinMergeContextMenuPackage.msix

Are there any others that could also be removed?

**Comment by sdottaka** [comment, marked as answer]
The files related to ShellExtension are:

- [x] Register.bat
- [x] RegisterPerUser.bat
- [x] RegisterWinMergeContextMenuPackage.bat
- [x] ShellExtensionU.dll
- [x] ShellExtensionX64.dll
- [x] UnRegister.bat
- [x] UnRegisterPerUser.bat
- [x] UnregisterWinMergeContextMenuPackage.bat
- [ ] vcomp140.dll
- [ ] WinMerge32BitPluginProxy.exe
- [x] WinMergeContextMenu.dll
- [x] WinMergeContextMenuPackage.msix

**Reply by Gitoffthelawn** [reply]
Thank you very much!
